### ISSUE TITLE
Using `useListBox` with selection mode `multiple` cause react warning: Cannot update a component while rendering a different component.

### ISSUE THREAD

**Issue by maksimsemenov** [adobe/react-spectrum#1611]
# 🐛 Bug Report 

When I use a `useListBox` hook with `selectionMode: "multiple"` in a controlled input pattern, selecting an options leads to react trying to update parent component, during the rendering of the listbox component:
```
Warning: Cannot update a component (`Control`) while rendering a different component (`Sectional`).
```

## 🤔 Expected Behavior
It shouldn't do that.

## 😯 Current Behavior

Please see the [reproduction repository](https://github.com/maksimsemenov/react-aria-state-bug) for the details.

## 💻 Code Sample
[Reproduction repository](https://github.com/maksimsemenov/react-aria-state-bug)

Steps to reproduce:
- Run the app
- Click on one of the items
- Notice the error/warning in the console

Selecting an option with a keyboard does not produce the error. Also, using the `selectionMode: "single"` does not produce an error either.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| @react-aria/focus| ^3.2.3
| @react-aria/listbox| ^3.2.4
| @react-aria/utils| ^3.6.0
| @react-stately/collections| ^3.3.0
| @react-stately/list| ^3.2.2

**Comment by LFDanLu** [comment]
@maksimsemenov I was able to get rid of the warning by modifying the `days` state setup to the following:
```
const [days, setDays] = useState(new Set() as Set<React.Key>)
const handleDaysChange = (newDays: Set<React.Key>): void => {
	setDays(newDays)
}
```
Not entirely sure why the array had this issue though, I will need to dig.

**Comment by LFDanLu** [comment]
@maksimsemenov Reproduced this issue locally, the changes to `useControlledState` in this [PR](https://github.com/adobe/react-spectrum/pull/1375) fixes it. You maybe able to try that change yourself if you patch the package locally

**Comment by maksimsemenov** [comment]
@LFDanLu sorry, it took me so long to check the fix. It works now. Thanks for fixing it!

**Comment by LFDanLu** [comment]
Great to hear! I'll see if we can get that PR merged sooner rather than later hopefully.

**Comment by akeohanePing** [comment]
@LFDanLu Hello there, is there any update on if this fix is going to be merged in? Thanks!

**Comment by snowystinger** [comment]
addressed by #2320 where we'll be removing our usage of useControlledState callbacks
will close this issue once those warnings are gone

**Comment by snowystinger** [comment]
We've finally merged this, #2304 It'll be available starting in the nightlies. Let us know if you find anything wrong with it.
